### DISCUSSION TITLE
Announcing anvil-zksync: New Name, Familiar Feel, and Exciting Upcoming Features

### DISCUSSION THREAD

**Discussion by dutterbutter** [zkSync-Community-Hub/zksync-developers#819] [Announcements]
Hey everyone! 👋

We’ve got news: **`era-test-node` is now `anvil-zksync` !**

**Why the change?**

We want to give developers a familiar, Anvil-like experience—matching APIs, CLI parity, and a smoother workflow. Starting now, `anvil-zksync` will be bundled with `foundry-zksync` for easy setup. 

**How to Get Started:**

Just update `foundry-zksync` using the [installer guide](https://foundry-book.zksync.io/getting-started/installation) and run:

```bash
anvil-zksync
```

**Documentation:**

Check out the [foundry-zksync-book](https://foundry-book.zksync.io/) for  `anvil-zksync` reference documentation.

**Release Notes:**

See what’s new: [v0.2.1 Release Notes](https://github.com/matter-labs/era-test-node/releases/tag/v0.2.1).

---

### ⚠️ Breaking Changes

**Asynchronous Transaction Execution**

Transactions are no longer executed synchronously. An example - previously `await sendTransaction()` guaranteed a finalized transaction. Now, you must wait for the transaction receipt before assuming it’s fully mined and validated.

**Example Fix:**

**Before:**

```tsx
const txReceipt = await greeter.setGreeting("Luke Skywalker");
const details = await provider.send("zks_getTransactionDetails", [txReceipt.hash]);
```

**After:**

```tsx
const txResponse = await greeter.setGreeting("Luke Skywalker");
const txReceipt = await txResponse.wait();
const details = await provider.send("zks_getTransactionDetails", [txReceipt.transactionHash]);
```

**Version Pinning & CI Updates**

- For CI setups, use [anvil-zksync-action](https://github.com/dutterbutter/anvil-zksync-action) for out-of-the-box support for `anvil-zksync` releases
- If you need older `era-test-node` versions, you can still use [era-test-node-action](https://github.com/dutterbutter/era-test-node-action) which will support some releases which can be viewed [here](https://github.com/matter-labs/era-test-node/releases).
- `@matterlabs/hardhat-zksync-node@v1.2.0` and below remains on `era-test-node:v0.1.0-alpha.36`. From `v1.3.0` forward, it will make use of `anvil-zksync`.

---

**Want to Contribute or Stay Updated?**

Check out the repo: [matter-labs/anvil-zksync](https://github.com/matter-labs/anvil-zksync)

Enjoy the improved developer experience! Have questions or issues? Drop a comment. Let’s build something awesome together!

**Comment by lutr0** [comment]
Love it!
Created an issue as the installation guide does not work on Linux: https://github.com/matter-labs/foundry-zksync/issues/777

**Reply by dutterbutter** [reply]
Will be resolved here: https://github.com/matter-labs/foundry-zksync/pull/780

**Reply by dutterbutter** [reply]
@lutr0 please try again, should be resolved.

**Reply by lutr0** [reply]
Appreciated!
I had to redownload the install script but it worked
